pos_mail_receipt/static/src/js/screens.js

```diff
@@ -48,9 +48,16 @@ odoo.define("pos_mail_receipt.screens", function (require) {
             var self = this;
             options = options || {};
             var timeout = typeof options.timeout === 'number' ? options.timeout : 7500;
-            this.$('.button.email').addClass("highlight");
+            var timeout_db_sync = typeof options.timeout_db_sync === 'number' ? options.timeout : 5000;
+            var i = 0
+            var waiting_interval = 1000
             while (self.pos.get("synch").state == "connecting") {
-                await sleep(1000);
+                await sleep(waiting_interval);
+                i = i + waiting_interval
+                if (i >= timeout_db_sync){
+                    self.gui.show_popup('error', _t('Timeout syncronising the database'));
+                    return
+                }
             }
             self.lock_screen(true);
```

Comment:
To be sure I understand, do you mean adding *another* message if the state is `connecting`, before waiting for the timeout ? I am affraid that this could trigger too often...

Reply:
Ah, GitHub UI is doing what GitHub UI does. I'm looking exclusively at the line `self.lock_screen(true);`. That line should be moved up :)

Reply:
Ah I see, it's done :)